src/plugins/embeddable/public/lib/embeddables/embeddable.tsx

```diff
@@ -131,6 +132,28 @@ export abstract class Embeddable<
     return this.output;
   }
 
+  public async getExplicitInputIsEqual(lastInput: Partial<TEmbeddableInput>): Promise<boolean> {
+    const currentInput = this.getExplicitInput();
+    return (
+      genericEmbeddableInputIsEqual(lastInput, currentInput) &&
+      fastIsEqual(omitGenericEmbeddableInput(lastInput), omitGenericEmbeddableInput(currentInput))
+    );
+  }
+
+  public getExplicitInput() {
+    if (this.getRoot().isContainer) {
+      return (
+        ((this.getRoot() as IContainer).getInput().panels?.[this.id]
+          ?.explicitInput as TEmbeddableInput) ?? this.getInput()
```

Comment:
I am not sure if this is logically correct to fallback to the whole input. Should we fallback to empty input instead? 

```suggestion
          ?.explicitInput as TEmbeddableInput) ?? {}
```

Reply:
This is a bit tricky, but _explicit_ input is the embeddables **own**, non-inherited input. If the embeddable doesn't have a parent, then _all_ of it's input is its own explicit input.